packages/react-data-grid/src/masks/InteractionMasks.tsx

```diff
@@ -0,0 +1,750 @@
+    this.setState(({ draggedPosition }) => {
+      if (draggedPosition) {
+        return { draggedPosition: { ...draggedPosition, overRowIdx } };
+      }
+      return null;
+    });
+  }
+
+  handleDragEnd = () => {
+    const { draggedPosition } = this.state;
+    if (draggedPosition === null) return;
+
+    const { rowIdx, overRowIdx } = draggedPosition;
+    const { columns, onGridRowsUpdated, rowGetter } = this.props;
+    const column = getSelectedColumn({ selectedPosition: draggedPosition, columns });
+    const value = getSelectedCellValue({ selectedPosition: draggedPosition, columns, rowGetter });
+    const cellKey = column.key;
+    const fromRow = rowIdx < overRowIdx ? rowIdx : overRowIdx;
+    const toRow = rowIdx > overRowIdx ? rowIdx : overRowIdx;
+
+    onGridRowsUpdated(cellKey, fromRow, toRow, { [cellKey]: value }, UpdateActions.CELL_DRAG);
+
+    this.setState({
+      draggedPosition: null
+    });
+  };
+
+  onDragHandleDoubleClick = (): void => {
+    const { onDragHandleDoubleClick, rowGetter } = this.props;
+    const { selectedPosition } = this.state;
+    const { idx, rowIdx } = selectedPosition;
+    const rowData = getSelectedRow({ selectedPosition, rowGetter });
+    onDragHandleDoubleClick({ idx, rowIdx, rowData });
+  };
+
+  onCommit = (...args: unknown[]): void => {
+    this.props.onCommit(...args);
+    this.closeEditor();
+  };
+
+  onCommitCancel = (): void => {
+    this.closeEditor();
+  };
+
+  getSelectedDimensions = (selectedPosition: Position, useGridColumns?: boolean): Dimension => {
+    const { scrollLeft, getRowHeight, getRowTop, getRowColumns, columns: gridColumns } = this.props;
+    const columns = useGridColumns ? gridColumns : getRowColumns(selectedPosition.rowIdx);
+    const top = getRowTop(selectedPosition.rowIdx);
+    const rowHeight = getRowHeight(selectedPosition.rowIdx);
+    const dimension = getSelectedDimensions({ selectedPosition, columns, scrollLeft, rowHeight });
+    dimension.top = top;
+    return dimension;
+  };
+
+  renderSingleCellSelectView() {
+    return (
+      !this.state.isEditorEnabled && this.isGridSelected() && (
+        <SelectionMask
+          {...this.getSelectedDimensions(this.state.selectedPosition, true)}
+          ref={this.selectionMask}
+        >
+          {this.isDragEnabled() && (
+            <DragHandle
+              onDragStart={this.handleDragStart}
+              onDragEnd={this.handleDragEnd}
+              onDoubleClick={this.onDragHandleDoubleClick}
+            />
+          )}
+        </SelectionMask>
+      )
+    );
+  }
+
+  renderCellRangeSelectView() {
+    const { columns, rowHeight } = this.props;
+    return (
+      <>
+        <SelectionRangeMask
+          {...getSelectedRangeDimensions({ selectedRange: this.state.selectedRange, columns, rowHeight })}
+        />
+        <SelectionMask
+          {...this.getSelectedDimensions(this.state.selectedRange.startCell, true)}
```

Comment:
I think we can probably use `this.state.selectedPosition` as `startCell` is same  as `selectedPosition`

https://github.com/adazzle/react-data-grid/blob/9a612b56c32da4dbbbc2761ccda36d0ab2ee61dd/packages/react-data-grid/src/masks/InteractionMasks.tsx#L516

Reply:
Done, hope it works.